vmm/src/device_manager.rs

```diff
@@ -563,22 +606,30 @@ impl DeviceManager {
                 let block = match image_type {
                     ImageType::Raw => {
                         let raw_img = vm_virtio::RawFile::new(raw_img);
-                        let dev =
-                            vm_virtio::Block::new(raw_img, disk_cfg.path.clone(), false, false)
-                                .map_err(DeviceManagerError::CreateVirtioBlock)?;
+                        let dev = vm_virtio::Block::new(
+                            raw_img,
+                            disk_cfg.path.clone(),
+                            false,
+                            disk_cfg.iommu,
```

Comment:
What happens if a user turns it on for a disk and forgets to turn it on globally?

Reply:
The virtio feature flag is enabled in this case, but because the virtio-iommu is not created, it has strictly no impact on the way the device works, with no impact on performance.